lib/upgradedb/postgres.2021040900.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ *  LMS version 1.11-git
+ *
+ *  (C) Copyright 2001-2021 LMS Developers
+ *
+ *  Please, see the doc/AUTHORS for more information about authors!
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License Version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
+ *  USA.
+ *
+ *  $Id$
+ */
+
+$this->BeginTrans();
+
+$this->Execute("ALTER TABLE rttickets ALTER COLUMN priority SET DEFAULT NULL");
+$this->Execute("ALTER TABLE rttickets ALTER COLUMN priority DROP NOT NULL");
+
```

Comment:
Tu bym na wszelki wypadek zmienił kolejność - najpierw SET DEFAULT NULL, gdy jeszcze przez chwilę jest niedozwolone NULL jest ciut niebezpieczne.

Reply:
Zastanowiłem się głębiej nad tym - ta druga linijka jest zupełnie niepotrzebna.
Postgres usunie ogranicznik "NOT NULL" gdy ustawimy "Default null" - co jest logiczne.
f08e883ea24fe7ead253a503127c4d1a6779613d